### ISSUE TITLE
Error during build process in hash-for-dep module

### ISSUE THREAD

**Issue by dennismende** [ember-cli/ember-cli#7390]
Output from `ember version --verbose && npm --version && yarn --version`:
```
ember-cli: 2.16.2
http_parser: 2.7.0
node: 6.11.3
v8: 5.1.281.107
uv: 1.11.0
zlib: 1.2.11
ares: 1.10.1-DEV
icu: 58.2
modules: 48
openssl: 1.0.2l
os: darwin x64
4.6.1
```

I get the following error shortly before the build breaks:
```
Cannot read property 'name' of null        
TypeError: Cannot read property 'name' of null        
at again (/cache/node_modules/hash-for-dep/lib/deps-for.js:19:25)        
at /cache/node_modules/hash-for-dep/lib/deps-for.js:27:7        
at Array.forEach (native)        
 at again (/cache/node_modules/hash-for-dep/lib/deps-for.js:26:55)        
at /cache/node_modules/hash-for-dep/lib/deps-for.js:27:7        
at Array.forEach (native)        
at again (/cache/node_modules/hash-for-dep/lib/deps-for.js:26:55)        
at /cache/node_modules/hash-for-dep/lib/deps-for.js:27:7        
at Array.forEach (native)        
at again (/cache/node_modules/hash-for-dep/lib/deps-for.js:26:55)
```

**Comment by rwjblue** [comment]
Hmm, this is almost certainly an issue in stefanpenner/hash-for-dep (not here). Mind filing this issue over there?

**Comment by rwjblue** [comment]
This seems like a scenario that I did not handle properly in https://github.com/stefanpenner/hash-for-dep/pull/38.

**Comment by dennismende** [comment]
@rwjblue Do you have a rough estimation when this could be fixed? Thank you very much.

**Comment by rwjblue** [comment]
Absolutely! It’s most likely about 30 minutes to an hour away from publishing the fix once someone submits it. 😆

**Comment by dennismende** [comment]
😆
I'm trying to test where the error comes from but till now no luck.

**Comment by frankjmattia** [comment]
I'm sorry for the "me too" but I was just bitten by this while setting up a new ember app. One thing I noticed is this only happens for me when using yarn.

```
frankjmattia@lappy-i686 % ember new testapp --yarn && cd testapp && ember serve
Yarn: Installed dependencies
Successfully initialized git.
Cannot read property 'name' of null
TypeError: Cannot read property 'name' of null
    at again (/Users/frankjmattia/testapp/node_modules/hash-for-dep/lib/deps-for.js:19:26)
    at /Users/frankjmattia/testapp/node_modules/hash-for-dep/lib/deps-for.js:27:7
    at Array.forEach (<anonymous>)
    at again (/Users/frankjmattia/testapp/node_modules/hash-for-dep/lib/deps-for.js:26:55)
    at /Users/frankjmattia/testapp/node_modules/hash-for-dep/lib/deps-for.js:27:7
    at Array.forEach (<anonymous>)
    at again (/Users/frankjmattia/testapp/node_modules/hash-for-dep/lib/deps-for.js:26:55)
    at /Users/frankjmattia/testapp/node_modules/hash-for-dep/lib/deps-for.js:27:7
    at Array.forEach (<anonymous>)
    at again (/Users/frankjmattia/testapp/node_modules/hash-for-dep/lib/deps-for.js:26:55)
```

If I drop `--yarn` from `ember new` everything goes swimmingly.

For completeness sake, here is the output of `ember version --verbose && npm --version && yarn --version`:
```
ember-cli: 2.16.2
http_parser: 2.7.0
node: 8.7.0
v8: 6.1.534.42
uv: 1.15.0
zlib: 1.2.11
ares: 1.10.1-DEV
modules: 57
nghttp2: 1.25.0
openssl: 1.0.2l
icu: 59.1
unicode: 9.0
cldr: 31.0.1
tz: 2017b
os: darwin x64
5.5.1
1.2.1
```

**Comment by frankjmattia** [comment]
I suppose if I had a better understanding of things I would have figured to just delete my `node_modules` folder and run `yarn install` (without `--ignore-optional`) a little sooner.

The comment from @rwjblue about stefanpenner/hash-for-dep#38 probably should have been the ah-ha moment but no... I spent an hour or so digging through what happens when you actually run `ember new` 😆 .

Good learning experience at least. Hopefully someday I'll contribute something tangible, but until then I can only offer my gratitude. 🙏

**Comment by rwjblue** [comment]
@frankjmattia - In stefanpenner/hash-for-dep#38 I made `pkg` able to return `null` when a package is not present, however I missed [this usage](https://github.com/stefanpenner/hash-for-dep/blob/master/lib/deps-for.js#L18).  The fix should likely be adding an early return there when `thePackage` is `null`, along with a previously failing test...

**Comment by frankjmattia** [comment]
Yessir, a simple `if (!thePackage) { return; }` does solve the problem. I'm trying to write a failing test case (and still learning my way around) but adding an optional, non-existent dep doesn't seem to trigger a failure 😞 .

Should this issue be closed and a new one filed with hash-for-dep?

**Comment by rwjblue** [comment]
Seems reasonable to create a second issue in hash-for-dep , but it’s also good to keep this issue around for folks searching here...

**Comment by rwjblue** [comment]
Fixed by @dennismende, and released as hash-for-dep@1.2.2.
